backend/rs/ic_vetkeys/src/utils/mod.rs

```diff
@@ -218,14 +209,19 @@ impl MasterPublicKey {
     /// Return the hardcoded master public key used on IC
     ///
     /// This allows performing public key derivation offline
-    pub fn production_key(key_id: MasterPublicKeyId) -> Self {
-        match key_id {
-            MasterPublicKeyId::Key1 => Self { point: *G2_KEY_1 },
-            MasterPublicKeyId::TestKey1 => Self {
-                point: *G2_TEST_KEY_1,
-            },
+    ///
+    /// Returns None if the provided key_id is not known
+    pub fn from_key_id(key_id: &VetKDKeyId) -> Option<Self> {
```

Comment:
It is still unclear if we will support pocket-ic keys. In case we do, given that in pocket-ic the keys have the same names, `VetKDKeyId` alone isn't sufficient in telling if the user wants the mainnet key or the pocket-ic key. So then we either have to switch back to an enum with variants for mainnet keys (e.g., `Mainnet_Bls12_381_G2_Key1`) and for pocket-ic keys (e.g., `PocketIc_Bls12_381_G2_Key1`), or we use different methods for instantiating mainnet keys (e.g., `from_mainnet_key`) and pocket-ic keys (`from_pocket_ic_key`). The latter seems somewhat preferred, given that then we can use the `VetKDKeyId` struct as parameter.

Given this, we could name this method `from_mainnet_key` now, and then have the option to add pocket-ic support in the future without a breaking change.

Reply:
👍